src/v/cluster/types.h

```diff
@@ -1607,4 +1618,19 @@ struct adl<cluster::prepare_tx_reply> {
         return cluster::prepare_tx_reply{ec};
     }
 };
+
+template<>
+struct adl<cluster::begin_tx_request> {
+    void to(iobuf& out, cluster::begin_tx_request&& r) {
+        serialize(
+          out, std::move(r.ntp), r.pid, r.tx_seq, r.transaction_timeout_ms);
+    }
+    cluster::begin_tx_request from(iobuf_parser& in) {
+        auto ntp = adl<model::ntp>{}.from(in);
+        auto pid = adl<model::producer_identity>{}.from(in);
+        auto tx_seq = adl<model::tx_seq>{}.from(in);
+        auto timeout = adl<model::timeout_clock::duration>{}.from(in);
```

Comment:
wondering if we should use duration that is not seastar clock dependent, as we know they change that from time to time

Reply:
yes, but i don't think a PR like should be changing it. presumably there are more PRs coming that will convert the whole tree?